community/bolt/src/main/java/org/neo4j/bolt/v1/transport/ChunkedOutput.java

```diff
@@ -186,7 +186,9 @@ private void ensure( int size ) throws IOException
         assert size <= maxChunkSize : size + " > " + maxChunkSize;
         if ( closed.get() )
         {
-            throw new PackOutputClosedException( "Network channel towards " + channel.remoteAddress() + " is closed. " + "Client has probably been stopped." );
+            throw new PackOutputClosedException(
+                    String.format( "Network channel towards %s is closed. Client has probably been stopped.",
+                            channel.remoteAddress() ), String.format( "%s", channel.remoteAddress() ) );
```

Comment:
Maybe just `channel.remoteAddress().toString()`? :)

Reply:
So, agree - I started with that - but it turned out there were a few tests where `channel.remoteAddress()` could be null. I was gonna just change the tests, but thought it'd be safer to just do the above in case there's some crazy condition where Netty returns null for `remoteAddress`; this way it'll handle `null` fine.